assets/js/theme/custom/wishlist-modal.js

```diff
@@ -0,0 +1,36 @@
+import modalFactory from '../global/modal'
+
+export default function wishlistModal() {
+    const selectors = {
+        wishlist: '.js-guest-wishlist-add',
+    }
+
+    const wishlistButtons = document.querySelectorAll(selectors.wishlist)
+    wishlistButtons?.forEach((button) => {
+        const title = button.getAttribute('data-modal-title')
+        const message = button.getAttribute('data-modal-message')
+        const login = button.getAttribute('data-modal-login')
+        const createAccount = button.getAttribute('data-modal-create-account')
+
+        if (button) {
+            button.addEventListener('click', (ev) => {
+                ev.preventDefault()
+                const modal = modalFactory('.js-preview-modal')
+                if (modal) {
+                    modal.open()
+
+                    const contentHTML = `
+                    <h1 class="c-modal__title">${title}</h1>
```

Comment:
Ok, it does solve the multi-language issue (I give you that 😅 ), but do we really need to inject it via js? Why can't we have a modal with the content inside html and just toggle its visibility via js? 🤔

Reply:
I didn't want to add another modal element in the HTML and the way I'm doing it now is just using what we already have

Reply:
@aleksandraworhacz Because that is how the modal currently works and I wanted to keep it consistent. What if in the future we keep adding modals? We will just have hidden HTML content on the page at all times and we will trigger it's visibility via JS. Having one modal HTML div where we can add content to it will reduce DOM complexity and page load time. Also, having multiple html modals seems redundant to me

Reply:
Well, I won't argue on having it consistent cause you have a point on that. But having more complex DOM structure isn't an issue, actually it's simpler than modifying the structure with js, so in terms of complexity I'd say the less js code the better and faster.  

>  Also, having multiple html modals seems redundant to me
How is it different than having multiple modal files but with js code? I think it's easier to maintain the code if you have dedicated HTML files for HTML components than searching for HTML code inside js file only in specific cases. 

In the end you decide what approach work best, just leaving here my humble opinion :)